modules/core/src/lib/pick-layers.js

```diff
@@ -91,12 +91,8 @@ export function pickObject(
         })) ||
       NO_PICKED_OBJECT;
 
-    if (!pickInfo.pickedColor) {
-      break;
-    }
-
     // only exclude if we need to run picking again
-    if (i + 1 < depth) {
+    if (pickInfo.pickedColor && i + 1 < depth) {
```

Comment:
Maybe add a comment for next maintainer explaining when this condition happens, if it is related to deep picking etc?